attrs_strict/_error.py

```diff
@@ -37,6 +37,22 @@ def __str__(self):
         return self._render(error)
 
 
+class EmptyError(BadTypeError):
```

Comment:
isn't this a breaking change, moving from ValueError to BadTypeError?

Reply:
Hmm I think this is ok, since BadTypeError has ValueError in its `MRO`. You can still catch `ValueError` but raise `EmptyError` so I'd say it is backwards compatible.

Reply:
@gaborbernat what do you think of this ?